### ISSUE TITLE
Problem about load_hair

### ISSUE THREAD

**Issue by Hacetate** [c-he/perm#5]
When I load the .data using io,py,I encountwered this error:  ` strands = np.array(strands).reshape((num_strands, -1,3))
ValueError: cannot reshape array of size 10000 into shape (10000,newaxis,3)` .
The reason for this is that numpy reads an irregular list and append to strands list, which cannot convert to a reshapable Numpy Array.
So what I need to do is zeroize or something else?

**Comment by Hacetate** [comment]
The Raw `.data` of USC-Hairsalon dataset all have strand of around `10,000`

**Comment by c-he** [comment]
Oh I see, the original data you downloaded from [their website](https://huliwenkidkid.github.io/liwenhu.github.io/) right? These data contain some irregular strands which only have one point. You can download our processed data from this [OneDrive link](https://yaleedu-my.sharepoint.com/:u:/g/personal/chengan_he_yale_edu/EZUbewJcyzpCo7iMjwZr7MQBPr_gLmKFzP8_l5rQDbYioQ). We filtered those strands and aligned them to our head mesh.

**Comment by Hacetate** [comment]
Thanks for the resampled original data. Could you please provide the code for preprocessing irregular data? I found that the dataset of the original `USC-hairsalon` is all 10000x300. However, the data from `Hairnet` interpolation contains a large number of strands greater than 300, while others range from 100 to 200. To obtain a better training dataset, it should not be sufficient to simply truncate strand above 300 and delete strand less than 300.

**Comment by c-he** [comment]
Sorry for the late reply, do you mean the data you downloaded from [Hair20k](https://zhouyisjtu.github.io/project_hair/hair20k.html) cannot be properly loaded by `load_hair()`? If so, could you provide some sample data? We can look into it later.

**Comment by wrk226** [comment]
> Sorry for the late reply, do you mean the data you downloaded from [Hair20k](https://zhouyisjtu.github.io/project_hair/hair20k.html) cannot be properly loaded by `load_hair()`? If so, could you provide some sample data? We can look into it later.

In fact, most hair data cannot be successfully loaded using `load_hair()`, because each strand has a different length. That is, the imported strands are a list of lists with unequal lengths. For example, you can check the file:

 `full_placed\hair_placed\L_curly\data\head\strands00009_00026_10001.data`

In this file, the first strand has 381 points while the second strand has 117 points. When running:

`strands = np.array(strands).reshape((num_strands, -1, 3))`

an error is raised:

`ValueError: setting an array element with a sequence. The requested array has an inhomogeneous shape after 1 dimensions. The detected shape was (9936,) + inhomogeneous part.`

**Comment by c-he** [comment]
Can you try the Hair20k data downloaded form this link: https://yaleedu-my.sharepoint.com/:f:/g/personal/chengan_he_yale_edu/EvTqlju7SolPp8vJMgAqczABEAJKQBSt3HBN_Yn8WogOIg? I assume you are using the data downloaded from the Hair20K website, which may not be fully proprocessed.

**Comment by wrk226** [comment]
You're right. The data downloaded from the new link you sent has no issues at all. Thank you very much.
